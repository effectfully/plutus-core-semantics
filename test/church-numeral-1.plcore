(program

(module Foo (imported) (exported () ()) (local () ())
(define four
  (let (lam s (lam z [s z]))           one
  (let (lam s (lam z [s [s [s z]]]))   three
  (let (lam x (builtin addInt x 1)) addOne
  (builtin addInt [[one addOne] 0] [[three addOne] 0])))))

(define six
  (let (lam s (lam z [s [s [s z]]]))   three
  (let (lam x (builtin addInt x 1)) addOne
  (builtin addInt [[three addOne] 0] [[three addOne] 0]))))
)

)
