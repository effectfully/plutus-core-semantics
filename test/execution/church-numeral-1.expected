<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . four ) |-> 4 ( Foo . six ) |-> 6 </genv>
  <modules>
    <Module>
      <curr> 0 </curr>
      <name> Foo </name>
      <dataCons> .TyExpList </dataCons>
      <decl> .NameList </decl>
      <importedModules> .ModNmList </importedModules>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </Module>
  </modules>
</T>
